src/support/mtx_rw.c

```diff
@@ -183,15 +186,18 @@ __wt_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
         /*
          * Fast path: if there is no active writer, join the current group.
          */
-        for (old.u.v = l->u.v; old.u.s.current == old.u.s.next; old.u.v = l->u.v) {
-            new.u.v = old.u.v;
+        for (__wt_atomic_storev64(&old.u.v, __wt_atomic_loadv64(&l->u.v));
```

Comment:
This is next level gross. I did call to move the mtx_rw.c file across but I didn't not expect this level of gross.

Reply:
It's something I called out in WT-12343, but in this case I messed up a mechanical change. `new` and `old` are stack variables and don't need the atomics. Fixed in [0242fdf](https://github.com/wiredtiger/wiredtiger/pull/10236/commits/0242fdf671db575063c6d91211871d941b5dd8af)